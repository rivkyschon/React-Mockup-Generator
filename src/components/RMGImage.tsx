// src/components/RMGImage.tsx

import React from 'react';

interface RMGImageProps {
  src: string;
  alt: string;
  width?: string;
  height?: string;
}

const RMGImage: React.FC<RMGImageProps> = ({ src, alt, width, height }) => {
  return (
<img
      src={src}
      alt={alt}
      width={width}
      height={height}
      className="rounded shadow-lg"
    />
  );
};

export default RMGImage;
