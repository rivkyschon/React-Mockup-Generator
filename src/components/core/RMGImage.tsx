// src/components/RMGImage.tsx

import React from 'react';

interface RMGImageProps {
  src?: string;
  alt?: string;
  width?: string;
  height?: string;
}

const RMGImage: React.FC<RMGImageProps> = ({src, alt, width, height}) => {
  return (
    <div className="w-60 h-60 m-auto flex justify-center items-center bg-gray-200">
      <img
        className="mx-auto d-block rounded shadow"
        src={src || 'https://placehold.jp/150x150.png'}
        alt={alt}
      />
    </div>
  );
};

export default RMGImage;
