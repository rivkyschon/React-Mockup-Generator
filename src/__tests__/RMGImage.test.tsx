// src/components/__tests__/RMGImage.test.tsx

import React from 'react';
import {render} from '@testing-library/react';
import RMGImage from '../components/RMGImage';

describe('RMGImage', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<RMGImage src="test.jpg" alt="test image" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has the correct src and alt text', () => {
    const {getByAltText} = render(<RMGImage src="test.jpg" alt="test image" />);
    const image = getByAltText('test image');
    // expect(image).toBeInTheDocument();
    // expect(image).toHaveAttribute('src', 'test.jpg');
  });
});
