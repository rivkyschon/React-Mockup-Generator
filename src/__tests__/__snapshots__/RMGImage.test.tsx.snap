// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMGImage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-60 h-60 flex justify-center items-center bg-gray-200"
  >
    <img
      alt="test image"
      class="mx-auto d-block rounded shadow"
      src="test.jpg"
    />
  </div>
</DocumentFragment>
`;
